performance/query_processing.rst

```diff
@@ -53,6 +63,69 @@ The task tracker executor is well suited for long running, complex data warehous
+
+  SELECT page_id, count(distinct session_id)
+  FROM visits
+  WHERE page_id IN (
+    SELECT page_id
+    FROM visits
+    GROUP BY page_id
+    ORDER BY count(*) DESC
+    LIMIT 20
+  )
+  GROUP BY page_id;
+
+The real-time executor would like to run a fragment of this query against each shard by page_id, counting distinct session_ids, and combining the results on the coordinator. However the LIMIT in the subquery means the subquery cannot be executed as part of the fragment. By recursively planning the query Citus can run the subquery separately, push the results to all workers, run the main fragment query, and pull the results back to the coordinator. The "push-pull" design supports a subqueries like the one above.
+
+For example, the EXPLAIN output for the previous query looks like this:
+
+::
+
+  GroupAggregate  (cost=0.00..0.00 rows=0 width=0)
+    Group Key: remote_scan.page_id
+    ->  Sort  (cost=0.00..0.00 rows=0 width=0)
+      Sort Key: remote_scan.page_id
+      ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+        ->  Distributed Subplan 7_1
+          ->  Limit  (cost=0.00..0.00 rows=0 width=0)
+            ->  Sort  (cost=0.00..0.00 rows=0 width=0)
+              Sort Key: COALESCE((pg_catalog.sum((COALESCE((pg_catalog.sum(remote_scan.worker_column_2))::bigint, '0'::bigint))))::bigint, '0'::bigint) DESC
+              ->  HashAggregate  (cost=0.00..0.00 rows=0 width=0)
+                Group Key: remote_scan.page_id
+                ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
+                  Task Count: 32
+                  Tasks Shown: One of 32
+                  ->  Task
+                    Node: host=localhost port=5433 dbname=postgres
+                    ->  HashAggregate  (cost=674.25..1004.42 rows=33017 width=12)
+                      Group Key: page_id
+                      ->  Seq Scan on visits_102200 visits  (cost=0.00..509.17 rows=33017 width=4)
+        Task Count: 32
+        Tasks Shown: One of 32
+        ->  Task
+           Node: host=localhost port=5433 dbname=postgres
+           ->  HashAggregate  (cost=734.53..899.61 rows=16508 width=8)
+             Group Key: visits.page_id, visits.session_id
+             ->  Hash Join  (cost=17.00..651.99 rows=16508 width=8)
+               Hash Cond: (visits.page_id = intermediate_result.page_id)
+               ->  Seq Scan on visits_102200 visits  (cost=0.00..509.17 rows=33017 width=8)
+               ->  Hash  (cost=14.50..14.50 rows=200 width=4)
+                 ->  HashAggregate  (cost=12.50..14.50 rows=200 width=4)
+                   Group Key: intermediate_result.page_id
+                   ->  Function Scan on read_intermediate_result intermediate_result  (cost=0.00..10.00 rows=1000 width=4)
+
+We can see that Citus uses the real-time executor to first execute the subquery as "Distributed Subplan 7_1," then runs another query to do the ``count(distinct session_id)`` with a HashAggregate.
```

Comment:
Is that what people should look for then? The presence of something like "Distributed Sub Plan"? Or can we show snippets of the plan, and show how data is flowing a bit clearer?